frontend/src/constants.ts

```diff
@@ -0,0 +1,29 @@
+import axios from 'axios'
+
+export interface Project {
+  id: string
+  name: string
+  created_on: string
+  expire_on?: string
+}
+export interface User {
+  id: string
+  created_on: string
+}
+interface Environ {
+  NAUTILUS_WEBAPI: string
+}
+interface ConstantsInterface {
+  env: Promise<Environ>
+}
+
+export const Constants: ConstantsInterface = {
+  env: (async function () {
+    try {
+      const response = await axios.get<Environ>(`${import.meta.env.BASE_URL}environ.json`)
```

Comment:
`${import.meta.env.BASE_URL}` is confusing because it resolves to nothing. Either drop it or define `BASE_URL` correctly in `vite.config.ts`